backend/authentication/tests.py

```diff
@@ -24,3 +28,106 @@ def test_str_methods() -> None:
     assert str(user_resource) == str(user_resource.id)
     assert str(user_task) == str(user_task.id)
     assert str(user_topic) == str(user_topic.id)
+
+
+@pytest.mark.django_db
+def test_signup(client: Client) -> None:
+    """Test the signup function.
+
+    Scenarios:
+
+    1. Password strength fails
+    2. Password confirmation fails
+    3. User is created successfully
+       - Check response status code
+       - Check if user exists in the DB
+       - Check if user password is hashed
+         (simple check if the password in the DB is the same as the raw password)
+    4. User already exists / Username already exists
+    5. Different User with the same email already exists
+
+    """
+    # Setup
+    fake = Faker()
+    username = fake.name()
+    second_username = fake.name()
+    email = fake.email()
+    strong_password = fake.password(
+        length=12, special_chars=True, digits=True, upper_case=True
+    )
+    wrong_password_confirmed = fake.password(
+        length=12, special_chars=True, digits=True, upper_case=True
+    )
+    weak_password = fake.password(
+        length=8, special_chars=False, digits=False, upper_case=False
+    )
+
+    # 1. Password strength fails
+    response = client.post(
+        path="/v1/auth/signup/",
+        data={
+            "username": username,
+            "password": weak_password,
+            "password_confirmed": weak_password,
+            "email": email,
+        },
+    )
+
+    assert response.status_code == 400
+    assert not UserModel.objects.filter(username=username).exists()
+
+    # 2. Password confirmation fails
+    response = client.post(
+        path="/v1/auth/signup/",
+        data={
+            "username": username,
+            "password": strong_password,
+            "password_confirmed": wrong_password_confirmed,
+            "email": email,
+        },
+    )
+
+    assert response.status_code == 400
+    assert not UserModel.objects.filter(username=username).exists()
+
+    # 3. User is created successfully
+    response = client.post(
+        path="/v1/auth/signup/",
+        data={
+            "username": username,
+            "password": strong_password,
+            "password_confirmed": strong_password,
+            "email": email,
+        },
+    )
+
+    assert response.status_code == 201
+    assert UserModel.objects.filter(username=username).exists()
+    assert UserModel.objects.get(username=username).password != strong_password
```

Comment:
Checking in here as I'm assuming the test passed, but then to me it would make sense that this is `==`?

Reply:
@andrewtavis Thanks for review 😃. 

Since the `exists()` method will return `True` or `False`, it is not necessary to write `==`.

Reply:
What I mean is for `assert UserModel.objects.get(username=username).password != strong_password`. To me we should be asserting `==`?

Reply:
Ah ok, I see. Maybe I should have placed a comment above there 😅.

There I am checking if the password is hashed. When I retrieve the password from the DB it should be hashed and therefore not the same as the raw password anymore.